### DISCUSSION TITLE
How to properly implement transition on data change?

### DISCUSSION THREAD

**Discussion by PixeledCode** [mhkeller/layercake#186] [Q&A] [answered]
Hi, thanks for this amazing library. I had question regarding the animation of chart on data change. 
For a basic bar chart, my approach is to update `rect` width using `tweened`, which is pretty much similar to the example on docs. 
Problem is that the axis is updating faster than the rectangle and it's creating a glitchy effect, where bar will update a little and then transition. You can see it here: https://svelte-pages.vercel.app/iom-charts?gender=all&age_group=all try changing some filters

Here is the code if you are interested: https://github.com/PixeledCode/svelte-pages/tree/main/src/routes/iom-charts/Bar

Thanks

**Comment by mhkeller** [comment, marked as answer]
Sorry I missed this. Instead of setting the width from inside the component, it is a better design to change your top-level items that get passed to the `<LayerCake>` component and then let everything in the chart update, similar to the [small multiples](https://layercake.graphics/example/SmallMultiples) example where it tweens the domain and everything else flows from there. If you are able to put a simplified version of your example in a svelte REPL it would be easier to debug a bit.

Edit: See this discussion for an example: https://github.com/mhkeller/layercake/discussions/205
